pkg/clusteragent/autoscaling/workload/controller_test.go

```diff
@@ -349,3 +357,138 @@ func TestDatadogPodAutoscalerTargetingClusterAgentErrors(t *testing.T) {
 		})
 	}
 }
+
+func TestLeaderCreateDeleteLocalHeap(t *testing.T) {
```

Comment:
nit: It's more something like `TestPodAutoscalerObjectsLimit` than testing the heap itself.